python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -582,7 +582,7 @@ def writeMetadata(
                     ref = ref.unresolved()
                 self.butler.put(metadata, ref)
```

Comment:
@TallJimbo this fails if we don't unresolve the ref because of this test code for replace run in test_cmdLineFwk:

```python
        # re-run with --replace-run (--inputs is ignored, as long as it hasn't
        # changed)
        args.replace_run = True
        args.output_run = "output/run2"
        fwk.runPipeline(copy.deepcopy(qgraph), taskFactory, args)
```
and you get a resolved ref for run `output/run1` being put into `output/run2` and it failing because it wants to put it into `output/run1`. 

@andy-slac  Interestingly we also get another failure in the simplest possible test:

```
        qgraph = fwk.makeGraph(self.pipeline, args)
        # run whole thing
        fwk.runPipeline(qgraph, taskFactory, args)
```

where the problem is:

```
'output/20230330T204714Z' != 'output/20230330T204715Z'
```
with error: `No collection with name 'output/20230330T204714Z' found.` which makes me wonder how we are generating two different versions of that timestamped output run.

Reply:
Of course that second test failure with the timestamp is random because it depends on timing. I can also get `testSimpleQGraphClobberOutputs` to fail with the same timestamp mismatch as `testSimpleQGraph`.

Reply:
If we don't unresolve refs then in `runPipeline` we should probably take output run from QuantumGraph. But I'm not sure how this will interact with all other options.